### ISSUE TITLE
Cfn-lint ignore-checks not honouring order of precedence 

### ISSUE THREAD

**Issue by laughingguffly** [aws-cloudformation/cfn-lint#2401]
### CloudFormation Lint Version

0.66.0

### What operating system are you using?

Windows

### Describe the bug

According to the readme:

Precedence
cfn-lint applies configurations from several sources. The rules at lower levels are overridden by those at higher levels.

cfnlintrc configurations
Template Metadata configurations
CLI parameters

If I run cfn-lint myfile.yaml -i E3002

and my template has a E3030 warning that is supressed in a metadata block.  
Any inline overrides specified in templates are ignored if I choose -i at the command line.

### Expected behavior

according to the precedence ordering I would expect cfn-lint run against the example template not to return any warning. It does and returns E3030. 

### Reproduction template

Resources:
  myInstance:
    Type: AWS::EC2::Instance
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3030
    Properties:
      InstanceType: nt.x4superlarge
      ImageId: ami-abc1234

**Comment by kddejong** [comment]
I cannot replicate this issue and I'll need more details.

Its worth noting that template and resource level Metadata behave differently.  Template level Metadata, CLI, and cfnlintrc files all operate at the entire template level and work with inheritance.  Resource level metadata is considered a directive and only functions on that resource.  The rule in this case is still processed and the errors are filtered out after the fact.

```yaml
Resources:
  myInstance:
    Type: AWS::EC2::Instance
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
          - E3030
    Properties:
      InstanceType: nt.x4superlarge
      ImageId: ami-abc1234
```

**Comment by kddejong** [comment]
Going to close for this now.  I can re-open it after more details emerge.
